x11-packages/luanti/0002-fix-elf-directory-for-cross-compiling.patch

```diff
@@ -0,0 +1,14 @@
+Luanti seems to not have previously been exposed to cross-compiling executables
+for platforms other than SurfaceFlinger. This prevents the error
+"CMake Error: failed to create symbolic link '/minetest': Permission denied"
+--- a/src/CMakeLists.txt
++++ b/src/CMakeLists.txt
+@@ -569,7 +569,7 @@ endif()
+ 
+ # When cross-compiling assume the user doesn't want to run the executable anyway,
+ # otherwise place it in <source dir>/bin/ since Luanti can only run from there.
+-if(NOT CMAKE_CROSSCOMPILING)
++if(NOT THIS_IS_FOR_SURFACEFLINGER_IN_OWOKITTYS_OPINION)
```

Comment:
That one patch line raises several questions.
All of them some variation of "what?"

- What are we actually patching here?
- What compelled you to choose that as the variable name?
- What does this have to do with SurfaceFlinger?

Reply:
A better name for this change would be "DISABLING_THIS_BECAUSE_IT_IS_FOR_BUILDING_AN_APK" , and the idea of "SurfaceFlinger" originates from the fact that `.apk` files that launch ANativeWindow activities cannot be executed on devices where the program named `/system/bin/surfaceflinger` is not running in the background. 

On the other hand, unlike the preexisting build of Minetest that can be found on the F-Droid store, which is a `.apk` file, this build targets X11 and therefore it can hypothetically run connected to the X11 server from the package `tigervnc` inside termux-docker which is an environment where `/system/bin/surfaceflinger` is not present.

I will work on renaming this variable now.

Reply:
@TomJo2000 to explain an answer to your question "What are we actually patching here?" in a way that is hopefully complete, precise and does not leave out any details:

Normally, Luanti does not set the `EXECUTABLE_OUTPUT_PATH` variable when cross-compiling is detected, because the person who wrote that section assumed/knew that they do not officially cross-compile for any platforms that require `EXECUTABLE_OUTPUT_PATH`.

Later on elsewhere in their `CMakeLists.txt`, they have this logic: "if not ANDROID... then if UNIX... then create a symlink in the `EXECUTABLE_OUTPUT_PATH`"

https://github.com/minetest/minetest/blob/6a1d22b2c516abbb9ce9670dedff47451317706f/src/CMakeLists.txt#L1045-L1054

The error that I saved in the comment of the patch, is the error that happens if `EXECUTABLE_OUTPUT_PATH` does not exist when that linked point in the `CMakeLists.txt` is reached.